viewer/src/public/createRevealManager.ts

```diff
@@ -12,40 +12,79 @@ import { LocalModelDataClient } from '../utilities/networking/LocalModelDataClie
 import { RevealOptions } from './types';
 import { initMetrics } from '../utilities/metrics';
 import { RenderOptions } from '..';
+import { EffectRenderManager } from '../datamodels/cad/rendering/EffectRenderManager';
+import { CadMaterialManager } from '../datamodels/cad/CadMaterialManager';
+import { OccludingGeometryProvider } from '../datamodels/cad/sector/culling/OccludingGeometryProvider';
 
 /**
  * Used to create an instance of reveal manager that works with localhost.
+ * @param renderer
+ * @param scene
  * @param revealOptions
  * @returns RevealManager instance.
  */
-export function createLocalRevealManager(revealOptions: RevealOptions = {}): RevealManager<LocalModelIdentifier> {
+export function createLocalRevealManager(
+  renderer: THREE.WebGLRenderer,
+  scene: THREE.Scene,
+  revealOptions: RevealOptions = {}
+): RevealManager<LocalModelIdentifier> {
   const modelDataClient = new LocalModelDataClient();
-  return createRevealManager('local', modelDataClient, revealOptions);
+  return createRevealManager('local', modelDataClient, renderer, scene, revealOptions);
 }
 
 /**
  * Used to create an instance of reveal manager that works with the CDF.
  * @param client
+ * @param renderer
+ * @param scene
  * @param revealOptions
  */
 export function createCdfRevealManager(
   client: CogniteClient,
+  renderer: THREE.WebGLRenderer,
+  scene: THREE.Scene,
   revealOptions: RevealOptions = {}
 ): RevealManager<CdfModelIdentifier> {
   const modelDataClient = new CdfModelDataClient(client);
-  return createRevealManager(client.project, modelDataClient, revealOptions);
+  return createRevealManager(client.project, modelDataClient, renderer, scene, revealOptions);
+}
+
+class RenderAlreadyLoadedGeometryProvider implements OccludingGeometryProvider {
```

Comment:
This class does not belong here. Seems very off to have the class `RenderAlreadyLoadedGeometryProvider` seemingly randomly located in createRevealManager.ts